diff --git a/libtcmu.c b/libtcmu.c
index c75bb10..7ef5737 100644
--- a/libtcmu.c
+++ b/libtcmu.c
@@ -29,7 +29,7 @@
 #include "libtcmu_log.h"
 #include "libtcmu_priv.h"
 #include "scsi_defs.h"
-
+#include "overlaybd/photon/export.h"
 #define TCMU_NL_VERSION 2

 static struct nla_policy tcmu_attr_policy[TCMU_ATTR_MAX+1] = {
@@ -384,7 +384,7 @@ void tcmu_dev_flush_ring(struct tcmu_device *dev)

 	tcmu_dev_dbg(dev, "waiting for ring to clear\n");
 	while (mb->cmd_head != mb->cmd_tail)
-		usleep(50000);
+		photon_thread_usleep(50000);
 	tcmu_dev_dbg(dev, "ring clear\n");
 }
